modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee

```diff
@@ -626,6 +629,42 @@ class ExperimentBrowserController extends Backbone.View
 
 			window.open("/openExptInQueryTool?experiment=#{code}",'_blank')
 
+	handleGetLinkQueryToolClicked: => 
+			# Preparatory Logic to Get Code to Pass to Generate Link Route 
+			if @experimentController.model.get('lsLabels') not instanceof LabelList
+				@experimentController.model.set 'lsLabels',  new LabelList @experimentController.model.get('lsLabels')
+			subclass = @experimentController.model.get('subclass')
+			if window.conf.entity?.saveInitialsCorpName? and window.conf.entity.saveInitialsCorpName is true
+				if @experimentController.model.get('lsLabels').getLabelByTypeAndKind("corpName", subclass + ' corpName').length > 0
+					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('corpName', subclass + ' corpName')[0].get('labelText')
+				else if @experimentController.model.get('lsKind') is "study" and @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id').length > 0
+					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
+				else
+					code = @experimentController.model.get("codeName")
+			else if @experimentController.model.get('lsKind') is "study" and @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id').length > 0
+				code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
+			else
+				code = @experimentController.model.get('codeName')
+			# Add Generating Link Loading Mask 
+			@$('.bv_generatingLink').show()
+			# Call to Route to Get URL 
+			$.ajax
+				type: 'GET'
+				url: "/getLinkExptQueryTool?experiment=#{code}"
+				success: (response) => 
+					@$('.bv_getLinkResults').show()
```

Comment:
I think to satisfy both ACAS-272 and ACAS-277 you should merge the logic of `handleOpenInQueryToolClicked` and this `handleGetLinkQueryToolClicked`. Could we have both buttons trigger the same "get a link" logic, and then have the logic diverge to either open a new tab with that link or to display the copyable link?